### PR TITLE
Correct Input Field Description in Component Sizing Information for Multi-Speed DX Heating Coils

### LINKED ISSUE #10611: Incorrect Input Field Description in Component Sizing Information in EIO file for Multi-Speed DX Heating Coils
Issue overview
--------------
While investigating issue #9909 uncovered incorrect `Input Field Description` in Component Sizing Information in EIO and html output files for `Coil:Heating:DX:MultiSpeed` coil. This is a defect.

See the component sizing information output in the EIO shows an incorrect name; heating COP is used for flow rate for speed 2 and Speed 1 heating capacity is replaced with waste heat fraction.

Develop:
```
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, Design Size Speed 2 Gross Rated Heating COP [m3/s], 0.39225
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 2 Gross Rated Heating COP [m3/s], 0.49554
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, Design Size Speed 1 Rated Air Flow Rate [m3/s], 0.19613
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 1 Rated Air Flow Rate [m3/s], 0.39644
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 1 Rated Waste Heat Fraction of Power Input [W], 9760.00000
```
This is caused by incorrect use of Field Names indices.

The correct reporting should have been:

```
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, Design Size Speed 2 Rated Air Flow Rate [m3/s], 0.39225
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 2 Rated Air Flow Rate [m3/s], 0.49554
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, Design Size Speed 1 Rated Air Flow Rate [m3/s], 0.19613
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 1 Rated Air Flow Rate [m3/s], 0.39644
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 2 Gross Rated Heating Capacity [W], 9760.00000
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Any)
 - Version of EnergyPlus ([develop](https://github.com/NREL/EnergyPlus/commit/15f64911c898c7d5dd3465b2bd472fd107e7f148))

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #10611
 - Corrects the input field description in component sizing information in `EIO` and `html` output files for multi-speed dx heating coils. Text diffs are expected.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -8290,7 +8290,7 @@ void SizeDXCoil(EnergyPlusData &state, int const DXCoilNum)
             CompType = thisDXCoil.DXCoilType;
             // Sizing rated air flow rate
             if (Mode == thisDXCoil.NumOfSpeeds) {
-                FieldNum = 12 + (Mode - 1) * 5;
+                FieldNum = 12 + (Mode - 1) * 6;
```
Comment (Nigusse) [inline src/EnergyPlus/DXCoils.cc:8293]: Corrected indices to Field Names of multi-speed DX Heating Coils.

File: src/EnergyPlus/DXCoils.cc [hunk 2]
```diff
@@ -8313,7 +8313,7 @@ void SizeDXCoil(EnergyPlusData &state, int const DXCoilNum)
                     bPRINT = true;
                 }
             } else {
-                FieldNum = 12 + (Mode - 1) * 5;
+                FieldNum = 12 + (Mode - 1) * 6;
```
Comment (Nigusse) [inline src/EnergyPlus/DXCoils.cc:8316]: Same as above.

File: src/EnergyPlus/DXCoils.cc [hunk 3]
```diff
@@ -8407,7 +8407,7 @@ void SizeDXCoil(EnergyPlusData &state, int const DXCoilNum)
             if (Mode == thisDXCoil.NumOfSpeeds) {
                 SizingMethod = HVAC::HeatingCapacitySizing;
                 state.dataSize->DataFlowUsedForSizing = thisDXCoil.MSRatedAirVolFlowRate(Mode);
-                FieldNum = 10 + (Mode - 1) * 5;
+                FieldNum = 10 + (Mode - 1) * 6;
```
Comment (Nigusse) [inline src/EnergyPlus/DXCoils.cc:8410]: Same as above.

File: src/EnergyPlus/DXCoils.cc [hunk 4]
```diff
@@ -8448,7 +8448,7 @@ void SizeDXCoil(EnergyPlusData &state, int const DXCoilNum)
             } else {
                 PrintFlag = true;
                 SizingMethod = HVAC::HeatingCapacitySizing;
-                FieldNum = 10 + (Mode - 1) * 5;
+                FieldNum = 10 + (Mode - 1) * 6;
```
Comment (Nigusse) [inline src/EnergyPlus/DXCoils.cc:8451]: Same as above.

File: tst/EnergyPlus/unit/DXCoils.unit.cc [hunk 5]
```diff
@@ -107,7 +107,7 @@ TEST_F(EnergyPlusFixture, DXCoils_Test1)
 
     state->dataDXCoils->DXCoilNumericFields.allocate(state->dataDXCoils->NumDXCoils);
     state->dataDXCoils->DXCoilNumericFields(2).PerfMode.allocate(1);
-    state->dataDXCoils->DXCoilNumericFields(2).PerfMode(1).FieldNames.allocate(17);
+    state->dataDXCoils->DXCoilNumericFields(2).PerfMode(1).FieldNames.allocate(18);
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/DXCoils.unit.cc:110]: Modified array size allocation.

File: tst/EnergyPlus/unit/DXCoils.unit.cc [hunk 6]
```diff
@@ -4648,9 +4648,9 @@ TEST_F(EnergyPlusFixture, TestMultiSpeedCoilsAutoSizingOutput)
     EXPECT_NEAR(16365.95, state->dataDXCoils->DXCoil(2).MSRatedTotCap(1), 0.01);
     // Check EIO reporting
     const std::string htg_coil_eio_output =
-        R"EIO( Component Sizing Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, Design Size Speed 2 Gross Rated Heating COP [m3/s], 1.75000
+        R"EIO( Component Sizing Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, Design Size Speed 2 Rated Air Flow Rate [m3/s], 1.75000
  Component Sizing Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, Design Size Speed 1 Rated Air Flow Rate [m3/s], 0.87500
- Component Sizing Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, Design Size Speed 1 Rated Waste Heat Fraction of Power Input [W], 32731.91226
+ Component Sizing Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, Design Size Speed 2 Gross Rated Heating Capacity [W], 32731.91226
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/DXCoils.unit.cc:4653]: Modified the text to match the correct input field description in the eio output file.

**Review by Nigusse — COMMENTED** [review]
Code change walk through.
- This change corrects the indices of numeric field names of multi-speed DX heating coils. This update was probably missed when a new field was added for the HSPF2 calculation.

**Comment by Nigusse** [comment]
EIO file results comparison based on the defect file:

Component Sizing Information Before the Fix:

```
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, Design Size Speed 2 Gross Rated Heating COP [m3/s], 0.39225
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 2 Gross Rated Heating COP [m3/s], 0.49554
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 1 Rated Waste Heat Fraction of Power Input [W], 9760.00000
```

Component Sizing Information After the Fix:

```
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, Design Size Speed 2 Rated Air Flow Rate [m3/s], 0.39225
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 2 Rated Air Flow Rate [m3/s], 0.49554
Component Sizing Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, User-Specified Speed 2 Gross Rated Heating Capacity [W], 9760.00000
```

**Comment by Nigusse** [comment]
Html file results comparison based on the defect file:

Results before the fix:
image

Results after the fix:
image

**Comment by Nigusse** [comment]
Defect file:
[in.idf.txt](https://github.com/user-attachments/files/16314143/in.idf.txt)

**Comment by Nigusse** [comment]
Diffs are expected in the `Component Sizing Information` input field descriptions in the `EIO` and `html` files. This fix impacts 18 example files with the `Coil:Heating:DX:MultiSpeed` object. This PR is ready for review.

**Review by mjwitte — APPROVED** [review]
Changes look good (even is they are a bit awkward). eio and table diffs are as expected. This is good to go. Thanks @Nigusse

**Pull request merged**